src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LIST_SIZE = 10;
+    private int size;
+    private Object[] list;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_LIST_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            list = increase(list);
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == list.length) {
+            list = increase(list);
+        }
+        if ((index >= 0) && (index <= size)) {
+            Object[] temporary = new Object[list.length];
+            System.arraycopy(list, 0, temporary,0, index);
+            temporary[index] = value;
+            System.arraycopy(list, index, temporary,index + 1, size - index);
+            list = temporary;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Your index is over the size");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if ((index >= 0) && (index < size)) {
+            return (T)list[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Your index is over the size");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if ((index >= 0) && (index < size)) {
+            list[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Your index is over the size");
+        }
     }
```

Comment:
In both of these methods, you have similar logic. Let's move it to the separate private one

Reply:
I moved this logic to the new method - isValid(index).
Please check is it correct to use it like this?